framework/sessions/__init__.py

```diff
@@ -44,6 +47,26 @@ def add_key_to_url(url, scheme, key):
     parsed_redirect_url = parsed_url._replace(**replacements)
     return urlparse.urlunparse(parsed_redirect_url)
 
+def get_node_id(url):
+    try:
+        node_id = [x for x in url.split("/") if x][2]
+        # Avoid circular import
+        from website.models import Node
+        if Node.find(Q('_id', 'eq', node_id)).count() == 1:
```

Comment:
Loading an invalid guid just returns `None`; this can be simplified.

Also, it seems like there should be a better way of doing this.